server/src/common/extensions.ts

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { URI } from 'vscode-uri';
+import { normalizePath } from './pathUtils';
+
+/* eslint-disable @typescript-eslint/no-empty-function */
+
+String.prototype.uriToPath = function(this: string): string {
```

Comment:
Yikes. I really don't think we should be extending "string" in this manner. These should be helper functions that take a string as a parameter. Otherwise we're polluting the namespace for a built-in type.

Reply:
Up to you, but this is TS extension methods thing. Pretty common...

Reply:
Just because a language allows you to do something doesn't mean it's a good idea. This is really fragile. I also see no benefit to doing it this way — only downsides.